### DISCUSSION TITLE
Hid/unHide Nodes in a Scene

### DISCUSSION THREAD

**Discussion by ztef** [vsg-dev/VulkanSceneGraph#855] [Q&A] [answered]
How can i hide/unhide nodes from the view ?

Thanks

**Comment by robertosfield** [comment, marked as answer]
The [vsg::Switch](https://github.com/vsg-dev/VulkanSceneGraph/blob/master/include/vsg/nodes/Switch.h#L22) node is designed for switch off the traversal of it's children. Each child of the Switch has a Mask (a 64bit mask( that is combined with the Visitor's traversalMask and if it's not 0 then the child is traversed.

To switch on a child use vsg::MASK_ALL, and switch off a child use vsg::MASK_OFF.

**Reply by WdsPjm** [reply]
If I only want to hide or show child nodes, is vsg::Switch also applicable?

**Reply by robertosfield** [reply]
That's exactly what vsg::Switch is for.

**Comment by WdsPjm** [comment]
Thank you very much. You are correct. It has been working in my project.
